pprof_analysis.go

```diff
@@ -471,6 +472,23 @@ func readPprofFile(pprof_file string) (*profile.Profile, error) {
 		content = out.Bytes()
 		zr.Reset(nil)
 	}
+
+	// handle zstd compressed profiles (magic bytes 0x28B52FFD / 0xFD2FB528)
+	if len(content) >= 4 {
+		magic := uint32(content[0]) | uint32(content[1])<<8 | uint32(content[2])<<16 | uint32(content[3])<<24
+		if magic == 0xFD2FB528 || magic == 0x28B52FFD { // little or big endian check
```

Comment:
```suggestion
		if magic == 0xFD2FB528 {
```
[RFC8878](https://datatracker.ietf.org/doc/html/rfc8878#section-3.1.1) defines the magic for Zstd as always little endian.